ios/sound.mm

```diff
@@ -24,209 +24,275 @@
 #include "sound.h"
 #include <AVFoundation/AVFoundation.h>
 
+#define kOutputBus 0
+#define kInputBus  1
 
 /* Implementation *************************************************************/
-CSound::CSound ( void           (*fpNewProcessCallback) ( CVector<short>& psData, void* arg ),
+CSound::CSound ( void ( *fpNewProcessCallback ) ( CVector<short>& psData, void* arg ),
                  void*          arg,
                  const QString& strMIDISetup,
-                 const bool     ,
+                 const bool,
                  const QString& ) :
     CSoundBase ( "CoreAudio iOS", fpNewProcessCallback, arg, strMIDISetup ),
-    midiInPortRef ( static_cast<MIDIPortRef> ( NULL ) )
+    isInitialized ( false )
 {
-    NSError *audioSessionError = nil;
-  
-    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
-    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
-        if (granted) {
-            // ok
-        }
-    }];
-    [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
+    try
+    {
+        NSError* audioSessionError = nil;
 
-  //GetAvailableInOutDevices();
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
+        [[AVAudioSession sharedInstance] requestRecordPermission:^( BOOL granted ) {
+            if ( granted )
+            {
+                // ok
```

Comment:
Might want to handle the case where permissions are not granted?

Reply:
Yeah same as #1576, when permission is not granted, app continues to work, just without sound.
Maybe an improvement in CSoundBase: a signal to CClient for example.

Reply:
Doesn't need to be addressed here. Probably use another PR for that.